### PR TITLE
Server 10.3 Release Notes

### PR DESCRIPTION
**Pull request by pmaier1**
This is the draft for 10.3 release notes. Please check carefully on correctness and language, as always :)

Some doc links are still missing because not available yet.

THX!

### CODE REVIEW

File: modules/admin_manual/pages/release_notes.adoc [hunk 1]
```diff
@@ -24,6 +25,151 @@
...
+* It is not recommended to continue with the deprecated apps. However, if you want to do so, you can copy over the `files_videoplayer` directory from the `apps/` folder of the previous ownCloud Server directory and obtain `gallery` from the https://marketplace.owncloud.com/apps/gallery[ownCloud Marketplace].
+
+TIP: Please do not enable `gallery`/`files_videoplayer` and `files_mediaviewer` simultaneously as these apps are mutually exclusive.
+
+For more information on the Media Viewer, visit the https://doc.owncloud.com/server/admin_manual/installation/apps/mediaviewer/[ownCloud Documentation].
+
+=== OAuth2 and session handling improvements
+
+Server 10.3 comes with improvements for session handling with Redis to cope for issues encountered around duplicate session tokens making the ownCloud Clients loose their OAuth2 authorization from time to time and forcing users to reauthenticate. The session handling stability has been generally improved and support for Redis Session Locking has been introduced. If Redis is used, it is necessary to enable Session Locking to ensure that the mentioned issues will not occur anymore.
```
Comment (phil-davis) [inline modules/admin_manual/pages/release_notes.adoc]: s/loose/lose

File: modules/admin_manual/pages/release_notes.adoc [hunk 2]
```diff
@@ -24,6 +25,151 @@
...
+
+The extension to integrate S3 object storages as secondary storages (`files_external_s3`) has been updated, unbundled from ownCloud Server (was previously part of `files_external`) and released to the https://marketplace.owncloud.com/apps/files_external_s3[ownCloud Marketplace]. If you're using S3 external storage mounts, you have to conduct some steps to ensure continuous operation after upgrading to Server 10.3:
+
+* After the upgrade to Server 10.3 has finished successfully, keep the maintenance mode activated and install/enable `files_external_s3` (either manually or via the Market app) as the app is not bundled with ownCloud Server anymore.
+* After enabling `files_external_s3` (`occ app:enable files_external_s3`) it is necessary to run `occ upgrade` again to conduct config migrations. This migration step should run quickly.
+* If users were allowed to configure personal mount points before the upgrade, switch from maintenance mode into single user mode (`occ maintenance:singleuser --on`) and enable the option again by ticking the respective checkbox below "Allow users to mount external storage" (in Admin settings => Storage).
```
Comment (micbar) [inline modules/admin_manual/pages/release_notes.adoc]: Click "Amazon S3" Storage ( Sub item of "Enable External Storages")
Maybe Screenshot?

File: modules/admin_manual/pages/release_notes.adoc [hunk 3]
```diff
@@ -24,6 +25,151 @@
...
+
+=> **Link to dev docs**
+
+=== Other notable changes
+
+* The `previews_path` config option has been added to allow customization of the thumbnail storage path (by default those reside in the user storage). https://github.com/owncloud/core/pull/35131[#35131]
+* An Activity entry will now be shown when a share receiver unshares a share https://github.com/owncloud/core/pull/35193[#35193]
+* The WebUI experience on mobile devices has been improved https://github.com/owncloud/core/pull/35919[#35919] https://github.com/owncloud/core/pull/35813[#35813] https://github.com/owncloud/core/pull/35347[#35347] https://github.com/owncloud/core/pull/34803[#34803] 
+* Two new config.php options (`proxy` and `proxyuserpwd`) have been added to enable federation when an instance needs to go through an authenticated proxy to reach a federated instance. See `config.sample.php` and the **LINK** [ownCloud Documentation] for more information. https://github.com/owncloud/core/pull/35868[#35868]
```
Comment (micbar) [inline modules/admin_manual/pages/release_notes.adoc]: They are not new, but federation ignored them in the past, They were already working for the updater

File: modules/admin_manual/pages/release_notes.adoc [hunk 4]
```diff
@@ -24,6 +25,151 @@
...
+* Two-factor providers may now display custom challenge messages https://github.com/owncloud/core/issues/34848[#34848]
+* The theming capabilities have been improved by allowing HTML for `Name` and `LogoClaim`. Please check https://github.com/owncloud/theme-example/pull/7/files[the changes to  owncloud/theme-example] if you are interested in making use of this in your theme. https://github.com/owncloud/core/pull/35273[#35273]
+* A new Roles API has been added to allow clients to query the server for available permissions/roles for user/group sharing and public links. In future client releases this endpoint will be used to dynamically display roles/permissions depending on the server's capabilities. https://github.com/owncloud/core/pull/36024[#36024]
+Docs: https://github.com/owncloud/docs/issues/1857
+* A new, improved version of the "Advanced Sharing Permissions" API (v2) has been added to allow ownCloud apps to register additional permissions/restrictions in user/group sharing. Version 1 of the API is still available in parallel. https://github.com/owncloud/core/pull/35836[#35863]
```
Comment (micbar) [inline modules/admin_manual/pages/release_notes.adoc]: Maybe add the information that this is a JS Api

**Comment by micbar** [comment]
@pmaier1 Great Work on the Cron Topic. This very precise and well written!

File: modules/admin_manual/pages/release_notes.adoc [hunk 5]
```diff
@@ -24,6 +25,151 @@
...
+Server 10.3 comes with improvements for session handling with Redis to cope for issues encountered around duplicate session tokens making the ownCloud Clients loose their OAuth2 authorization from time to time and forcing users to reauthenticate. The session handling stability has been generally improved and support for Redis Session Locking has been introduced. If Redis is used, it is necessary to enable Session Locking to ensure that the mentioned issues will not occur anymore.
+
+* It is recommended to use Redis Session Locking (minimum required version for `php-redis` is `4.1.0`)
+* Enable Redis Session Locking by setting `redis.session.locking_enabled = 1` in `php.ini`
+
+TIP: Please note that Redis Session Locking is not supported in clustered Redis environments.
+
+TIP: If you do not use Redis at all, you do not have to apply changes.
+
+TIP: If your ownCloud deployment is based on the official Docker images or the Univention appliance, these changes have already been applied for you.
```
Comment (micbar) [inline modules/admin_manual/pages/release_notes.adoc]: hopefully, not tested yet.

Comment (settermjd) [reply modules/admin_manual/pages/release_notes.adoc]: This isn't guaranteed?

Comment (micbar) [reply modules/admin_manual/pages/release_notes.adoc]: Ubuntu 18.04 has php-redis v3
Docker containers have Ubuntu 18.04

Ubuntu 20.04 not ready.

File: modules/admin_manual/pages/release_notes.adoc [hunk 6]
```diff
@@ -24,6 +25,205 @@
...
+
+=== Changes to background job execution
+
+For code cleanup reasons, the execution of background jobs (e.g., for public link expiration, trash bin emptying, cleanup of old file versions) has been changed.
+
+**These changes require manual interaction in your installation:**
+
+* If you're using https://doc.owncloud.com/server/admin_manual/configuration/server/background_jobs_configuration.html#cron[System cron] to trigger background job execution, there is a new `occ` command (`occ system:cron`) which executes the background jobs. 
+  To make use of it, you have to change the entry in `crontab`. 
+  Instead of executing `cron.php` (e.g., `/usr/bin/php -f /path/to/your/owncloud/cron.php`), cron should use `occ system:cron` (e.g., `/usr/bin/php -f /path/to/your/owncloud/occ system:cron`).
```
Comment (settermjd) [inline modules/admin_manual/pages/release_notes.adoc]: @pmaier1, I'm running "10.3.0 alpha2" on my testing box and don't see `system:cron`; however, I do see `background:cron`. Is this the command that you're referencing?

File: modules/admin_manual/pages/release_notes.adoc [hunk 7]
```diff
@@ -24,6 +25,205 @@
...
+  https://github.com/owncloud/core/pull/35998[#35998]
+* The `share_folder` `config.php` option now also respects federated shares. 
+  https://github.com/owncloud/core/pull/35396[#35396]
+* The `user.min_search_length` config.php option now also respects federated users. 
+  https://github.com/owncloud/core/pull/35977[#35977]
+* Issues with database conversions using the `db:convert-type` occ command (e.g., SQLite to MySQL) have been fixed. 
+  This is still in an experimental state and should be tested thoroughly. 
+  Please provide feedback if you encounter issues. 
+  https://github.com/owncloud/core/pull/35390[#35390]
+* *File integrity checking has been improved to prevent issues:* If a checksum mismatch occurs after uploading a file, the uploaded file and its checksum is deleted to prepare for a clean re-upload.
```
Comment (pmaier1) [inline modules/admin_manual/pages/release_notes.adoc]: is there a reason for making some entries bold and some not? @settermjd

File: modules/admin_manual/pages/release_notes.adoc (hunk 4 above)
Comment (pmaier1) [reply modules/admin_manual/pages/release_notes.adoc]: done

File: modules/admin_manual/pages/release_notes.adoc (hunk 3 above)
Comment (pmaier1) [reply modules/admin_manual/pages/release_notes.adoc]: thx, changed

File: modules/admin_manual/pages/release_notes.adoc (hunk 7 above)
Comment (settermjd) [reply modules/admin_manual/pages/release_notes.adoc]: The ones that were bolded were done based on how you wrote those parts of the content.

File: modules/admin_manual/pages/release_notes.adoc [hunk 8]
```diff
@@ -24,6 +25,205 @@
...
+// TODO 
+* The config.php options `proxy` and `proxyuserpwd` will now be respected to enable federation when an instance needs to go through an authenticated proxy to reach a federated instance. See `config.sample.php` and the **LINK** [ownCloud Documentation] for more information. https://github.com/owncloud/core/pull/35868[#35868]
+
+https://github.com/owncloud/docs/issues/1869
+
+* The `occ files:scan` command is now case-insensitive for the userid https://github.com/owncloud/core/pull/35324[#35324]
+* A new `config.php` option (`dav.enable.tech_preview`) has been added to disable tech preview APIs by default https://github.com/owncloud/core/issues/36124[#36124]
+* [PHOENIX] Support for redirecting links to ownCloud Phoenix frontend has been added by introducing a new `config.php` option which stores the address where Phoenix is reachable (e.g., `'phoenix.baseUrl' => 'http://phoenix.example.tld:port'`) https://github.com/owncloud/core/pull/35819[#35819]
+* When loading groups, the performance of users has been improved https://github.com/owncloud/core/pull/35822[#35822]
```
Comment (pmaier1) [inline modules/admin_manual/pages/release_notes.adoc]: "The performance of users"? This has a different sense now, sounds wrong.

Comment (settermjd) [reply modules/admin_manual/pages/release_notes.adoc]: Reverted.

**Comment by settermjd** [comment]
> @settermjd
> Could you find the system:cron job

Yes. I overlooked in when I wrote my original comment.

File: modules/admin_manual/pages/release_notes.adoc [hunk 9]
```diff
@@ -95,18 +95,21 @@ For more information on the Media Viewer app, visit the https://doc.owncloud.com
 === OAuth2 and session handling improvements
 
 Server 10.3 comes with improvements for session handling with Redis.
-These are designed to cope with issues encountered around duplicate session tokens, which make the ownCloud Clients lose their OAuth2 authorization from time to time, and forcing users to re-authenticate. 
-The session handling stability has been generally improved, and support for Redis Session Locking has been introduced. 
+These are designed to cope with issues encountered around duplicate session tokens, which make the ownCloud Clients lose their OAuth2 authorization from time to time, and forcing users to re-authenticate.
```
Comment (settermjd) [inline modules/admin_manual/pages/release_notes.adoc]: The final clause in this sentence doesn't read correctly to me. I suggest changing it to ", which forces users to re-authenticate".

File: modules/admin_manual/pages/release_notes.adoc [hunk 10]
```diff
@@ -95,18 +95,21 @@ For more information on the Media Viewer app, visit the https://doc.owncloud.com
...
+
+The session handling in ownCloud 10.3.0 has been generally improved, making user and client sessions more stable now.
```
[the remark this answers is not in the document]
Comment (pmaier1) [reply modules/admin_manual/pages/release_notes.adoc]: removed, thx

File: modules/admin_manual/pages/release_notes.adoc [hunk 11]
```diff
@@ -97,14 +97,14 @@ For more information on the Media Viewer app, visit the https://doc.owncloud.com
...
 The session handling stability has been generally improved, and support for Redis Session Locking has been introduced. 
-If Redis is used, it is necessary to enable Session Locking to ensure that the mentioned issues no longer occur. 
+If Redis is used for session handling, it is necessary to enable Session Locking to ensure that the mentioned issues no longer occur. 
 
-* It is recommended to use Redis Session Locking (minimum required version for `php-redis` is `4.1.0`)
+* It is recommended to use Redis Session Locking if Redis is used for session handling (minimum required version for `php-redis` is `4.1.0`)
 * Enable Redis Session Locking by setting `redis.session.locking_enabled = 1` in `php.ini`
 
-TIP: Please note that Redis Session Locking is not supported in clustered Redis environments.
+TIP: If Redis is just used as a memcache or not in use at all, you do not have to apply changes.
```
[the remark this answers is not in the document]
Comment (pmaier1) [reply modules/admin_manual/pages/release_notes.adoc]: changed to "memory cache"

File: modules/admin_manual/pages/release_notes.adoc [hunk 12]
```diff
@@ -24,6 +25,193 @@
...
+* Enable Redis Session Locking by setting `redis.session.locking_enabled = 1` in `php.ini`
+
+TIP: If Redis is just used as a memory cache or not in use at all, you do not have to apply changes.
+
+TIP: Please note that Redis Session Locking is not supported in clustered Redis environments.
+
+TIP: If your ownCloud deployment is based on the official Docker images or the Univention appliance, you do not have to apply changes as Redis is not used for session handling (unless you configured it differently using ENV variables).
+
+// TODO
+**Docs?**
```
Comment (phil-davis) [inline modules/admin_manual/pages/release_notes.adoc]: There are still these `TODO` lines here - is there still something to do? Or delete these lines?

Comment (settermjd) [reply modules/admin_manual/pages/release_notes.adoc]: There should only be one left. They're Asciidoc comments, so they won't appear in the published documentation.

Comment (settermjd) [reply modules/admin_manual/pages/release_notes.adoc]: Just removed that one, as it doesn't appear to be necessary.

File: modules/admin_manual/pages/release_notes.adoc [hunk 13]
```diff
@@ -24,6 +25,190 @@
...
+Please get in contact with ownCloud Support if you're still using OpenStack SWIFT and want to upgrade.
+
+=== S3 object storage as secondary storage is now a separate app
+
+The extension to integrate S3 object storages as secondary storages (`files_external_s3`) has been updated, unbundled from ownCloud Server (was previously part of `files_external`) and released to the https://marketplace.owncloud.com/apps/files_external_s3[ownCloud Marketplace]. 
+If you're using S3 external storage mounts, you have to conduct some steps to ensure continuous operation after upgrading to Server 10.3:
+
+* After the upgrade to Server 10.3 has finished successfully, keep the maintenance mode activated and install/enable `files_external_s3` (either manually or via the Market app) as the app is not bundled with ownCloud Server anymore.
+* After enabling `files_external_s3` (`occ app:enable files_external_s3`) it is necessary to run `occ upgrade` again to conduct config migrations.
```
Comment (micbar) [inline modules/admin_manual/pages/release_notes.adoc]: This step is not necessary! We just tested it. The migration is run during app:enable.
Please remove the second `occ upgrade` step.

Comment (settermjd) [reply modules/admin_manual/pages/release_notes.adoc]: Shall do.

Comment (settermjd) [reply modules/admin_manual/pages/release_notes.adoc]: Removed.

**Comment by settermjd** [comment]
@micbar, all right to go now?

**Pull request merged**